File: ArtIsPain/Client/src/app/commands/albums/upsert-song-command.ts
import { Time } from "@angular/common";
import { IUpsertEntityCommand } from "../../_interfaces/i-upsert-entity-command";
import { IDynamicallyAdded } from "../../_interfaces/i-dynamically-added";

export class UpsertSongCommand implements IUpsertEntityCommand, IDynamicallyAdded {
  EntityId?: string;
  AlbumId?: string;
  Title: string;
  Order: number;
  Length: Time;
  IsEditMode = false;
  IsNewlyAdded: boolean;
}

File: ArtIsPain/Server/Commands/Album/UpsertSongCommand.cs
using System;
using ArtIsPain.Server.Commands;
using Server.ViewModels.Album.Song;

namespace Server.Commands.Album
{
    public class UpsertSongCommand : IUpsertEntityCommand<SongPreviewModel>
    {
        public Guid? EntityId { get; set; }

        public Guid AlbumId { get; set; }

        public string Title { get; set; }

        public int Order { get; set; }

        public TimeSpan Length { get; set; }
    }
}